mozilla/activity-stream
# feat(mc): #2249 Add Store, Actions, Reducers, and Redux

This patch adds a store, actions, reducers, and Redux to the system addon. 

## Code
An instance of`lib/Store.jsm` when `lib/ActivityStream.jsm` starts up. It inititializes a redux store, as well as a list of `Feed`s.

I've included a really simple one for now called `NewTabInit`, that just logs the initial state. When the `MessageManager` lands, it can properly be hooked up to send the initial state to the content page.

## Tests
I had to add a few things to help with tests, including:
- `rewire`, a babel plugin that allows us to rewrite `import` statements;
- `GlobalOverrider`, a utility to set and restore global variables (e.g. `Components.utils`). unfortunately `rewire` doesn't support globals.

As usual, run `testmc` to run the tests.

## Manually testing

Try running the code in Firefox and checking your console output. You should see the redux state logged by the `NewTabInit` Action subscriber.
